test/ResultTest.cpp

```diff
@@ -0,0 +1,585 @@
+  EXPECT_TRUE(result2.isFullyMaterialized());
+  EXPECT_THROW(result2.idTables(), ad_utility::Exception);
+}
+
+// _____________________________________________________________________________
+TEST(Result,
+     verifyAssertSortOrderIsRespectedThrowsWhenNotSortedAndSucceedsWhenSorted) {
+  if constexpr (!ad_utility::areExpensiveChecksEnabled) {
+    GTEST_SKIP_("Expensive checks are disabled, skipping test.");
+  }
+  auto idTable = makeIdTableFromVector({{1, 6, 0}, {2, 5, 0}, {3, 4, 0}});
+
+  EXPECT_NO_THROW((Result{idTable.clone(), {}, LocalVocab{}}));
+
+  for (auto& generator : getAllSubSplits(idTable)) {
+    Result result{std::move(generator), {}, LocalVocab{}};
+    EXPECT_NO_THROW(consumeGenerator(result.idTables()));
+  }
+
+  EXPECT_NO_THROW((Result{idTable.clone(), {0}, LocalVocab{}}));
+
+  for (auto& generator : getAllSubSplits(idTable)) {
+    Result result{std::move(generator), {0}, LocalVocab{}};
+    EXPECT_NO_THROW(consumeGenerator(result.idTables()));
+  }
+
+  EXPECT_NO_THROW((Result{idTable.clone(), {0, 1}, LocalVocab{}}));
+
+  for (auto& generator : getAllSubSplits(idTable)) {
+    Result result{std::move(generator), {0, 1}, LocalVocab{}};
+    EXPECT_NO_THROW(consumeGenerator(result.idTables()));
+  }
+
+  EXPECT_NO_THROW((Result{idTable.clone(), {2, 0}, LocalVocab{}}));
+
+  for (auto& generator : getAllSubSplits(idTable)) {
+    Result result{std::move(generator), {2, 0}, LocalVocab{}};
+    EXPECT_NO_THROW(consumeGenerator(result.idTables()));
+  }
+
+  AD_EXPECT_THROW_WITH_MESSAGE_AND_TYPE(
+      (Result{idTable.clone(), {1}, LocalVocab{}}),
+      ::testing::HasSubstr("compareRowsBySortColumns"), ad_utility::Exception);
+
+  for (auto& generator : getAllSubSplits(idTable)) {
+    Result result{std::move(generator), {1}, LocalVocab{}};
+    AD_EXPECT_THROW_WITH_MESSAGE_AND_TYPE(
+        consumeGenerator(result.idTables()),
+        ::testing::HasSubstr("compareRowsBySortColumns"),
+        ad_utility::Exception);
+  }
+
+  AD_EXPECT_THROW_WITH_MESSAGE_AND_TYPE(
+      (Result{idTable.clone(), {1, 0}, LocalVocab{}}),
+      ::testing::HasSubstr("compareRowsBySortColumns"), ad_utility::Exception);
+
+  for (auto& generator : getAllSubSplits(idTable)) {
+    Result result{std::move(generator), {1, 0}, LocalVocab{}};
+    AD_EXPECT_THROW_WITH_MESSAGE_AND_TYPE(
+        consumeGenerator(result.idTables()),
+        ::testing::HasSubstr("compareRowsBySortColumns"),
+        ad_utility::Exception);
+  }
+
+  AD_EXPECT_THROW_WITH_MESSAGE_AND_TYPE(
+      (Result{idTable.clone(), {2, 1}, LocalVocab{}}),
+      ::testing::HasSubstr("compareRowsBySortColumns"), ad_utility::Exception);
+
+  for (auto& generator : getAllSubSplits(idTable)) {
+    Result result{std::move(generator), {2, 1}, LocalVocab{}};
+    AD_EXPECT_THROW_WITH_MESSAGE_AND_TYPE(
+        consumeGenerator(result.idTables()),
+        ::testing::HasSubstr("compareRowsBySortColumns"),
+        ad_utility::Exception);
+  }
```

Comment:
Make this a helper function (best one with a `source_location` argument and a `generateLocationTrace`.